client/clientd/src/lib.rs

```diff
@@ -58,6 +58,17 @@ impl PendingResponse {
     }
 }
 
+#[derive(Serialize)]
+pub struct PegInAddressResponse {
+    pegin_address: bitcoin::Address,
+}
+
+impl PegInAddressResponse {
+    pub fn new(pegin_address: bitcoin::Address) -> Self {
```

Comment:
You could think about just making the member `pub`. Private members + constructors are mostly interesting in two cases:

1. You want to enforce an invariant (e.g. a number that can never be 0, you check that at construction time and then can rely on that fact in your code)
2. To allow changes to the internal structure later on without changing the API

Both isn't really the case here:
1. Any address is valid
2. `serde` already exposes the members in a (public?) API

Reply:
yes right